### PR TITLE
HDF5 1.10 fix various warnings

### PR DESCRIPTION
**Pull request by jhendersonHDF**

### CODE REVIEW

File: src/H5C.c [hunk 1]
```diff
@@ -103,6 +103,9 @@
 /* Local Typedefs */
 /******************/
 
+/* Alias for pointer to cache entry, for use when allocating sequences of them */
+typedef H5C_cache_entry_t *H5C_cache_entry_ptr_t;
```
Comment (jhendersonHDF) [inline src/H5C.c:107]: Changes in this file are to align with d5130bb57332ce0049302c5836bbf666b23513cd from develop

File: src/H5FDmpio.c [hunk 2]
```diff
@@ -1317,36 +1317,6 @@ H5FD__mpio_get_handle(H5FD_t *_file, hid_t H5_ATTR_UNUSED fapl, void **file_hand
...
- *
- * Returns:     Non-negative if succeed or negative if fails.
- *
- * Programmer:  John Mainzer
- *              April 4, 2017
- *
- *-------------------------------------------------------------------------
- */
-static herr_t
-H5FD_mpio__get_info(H5FD_t *_file, void **mpi_info)
```
Comment (jhendersonHDF) [inline src/H5FDmpio.c:1333]: Not used anywhere in 1.10

File: src/H5Fprivate.h [hunk 3]
```diff
@@ -188,14 +188,13 @@ typedef struct H5F_t H5F_t;
...
         (p)++;                                                                                               \
-        (i) |= ((int32_t)(*(p) & (unsigned)0xff) << 8);                                                      \
+        (i) |= ((int32_t)(*(p)&0xff) << 8);                                                                  \
         (p)++;                                                                                               \
-        (i) |= ((int32_t)(*(p) & (unsigned)0xff) << 16);                                                     \
+        (i) |= ((int32_t)(*(p)&0xff) << 16);                                                                 \
         (p)++;                                                                                               \
-        (i) |= ((int32_t)(((*(p) & (unsigned)0xff) << 24) |                                                  \
-                          ((*(p) & (unsigned)0x80) ? (unsigned)(~0xffffffff) : (unsigned)0x0)));             \
+        (i) |= ((int32_t)(((*(p) & (unsigned)0xff) << 24) | ((*(p)&0x80) ? ~0xffffffffULL : 0x0ULL)));       \
```
Comment (jhendersonHDF) [inline src/H5Fprivate.h:197]: Align with develop and eliminate many warnings about identical branches.

File: src/H5Fsuper.c [hunk 4]
```diff
@@ -518,12 +518,9 @@ H5F__super_read(H5F_t *f, H5P_genplist_t *fa_plist, hbool_t initial_read)
...
     if (!H5F_addr_eq(super_addr, sblock->base_addr)) {
-        /* Check if the superblock moved earlier in the file */
-        if (H5F_addr_lt(super_addr, sblock->base_addr))
-            udata.stored_eof -= (sblock->base_addr - super_addr);
-        else
-            /* The superblock moved later in the file */
-            udata.stored_eof += (super_addr - sblock->base_addr);
+        /* If the superblock moved in the file, adjust the EOF */
+        /* (Handles moving earlier & later) */
+        udata.stored_eof -= (sblock->base_addr - super_addr);
```
Comment (jhendersonHDF) [inline src/H5Fsuper.c:523]: This change aligns with 1ccbdfee589d9f56d71983506313879fa47c4e90 from develop

File: src/H5Pfapl.c [hunk 5]
```diff
@@ -1187,12 +1187,14 @@ H5P__file_driver_free(void *value)
 
                 /* Allow driver to free info or do it ourselves */
                 if (driver->fapl_free) {
-                    if ((driver->fapl_free)((void *)info->driver_info) < 0) /* Casting away const OK -QAK */
+                    /* Free the const pointer */
+                    /* Cast through uintptr_t to de-const memory */
+                    if ((driver->fapl_free)((void *)(uintptr_t)info->driver_info) < 0)
```
Comment (jhendersonHDF) [inline src/H5Pfapl.c:1192]: Ugly hack, but aligns with develop until we change it everywhere

File: src/H5SL.c [hunk 6]
```diff
@@ -880,10 +880,18 @@ H5SL_release_common(H5SL_t *slist, H5SL_operator_t op, void *op_data)
...
 
-        /* Call callback, if one is given */
+        /* Call callback, if one is given.
+         *
+         * Ignoring const here is fine as we only need the value to be const
+         * with respect to the list code, which should never modify the
+         * elements. The library code that is making use of the skip list
+         * container can do what it likes with the elements.
+         */
+        H5_GCC_DIAG_OFF("cast-qual")
```
Comment (jhendersonHDF) [inline src/H5SL.c:890]: Everything in this file is just aligning with develop

File: test/h5test.c [hunk 7]
```diff
@@ -561,25 +560,7 @@ h5_fixname_real(const char *base_name, hid_t fapl, const char *_suffix, char *fu
...
-                    env = HDF5_DRIVER;
-#endif
-                if (env && !HDstrcmp(env, "split")) {
-                    /* split VFD */
-                    suffix = NULL;
-                }
-                else {
-                    /* multi VFD */
-                    suffix = NULL;
-                }
```
Comment (jhendersonHDF) [inline test/h5test.c:582]: Checking HDF5_DRIVER when picking test filename suffix isn't used in 1.10.

**Review by derobins — APPROVED** [review]

**Pull request merged**